src/main/webapp/app/components/Tabs/CurationHistoryTab.tsx

```diff
@@ -0,0 +1,259 @@
+import constructTimeSeriesData, { formatLocation } from '../geneHistoryTooltip/gene-history-tooltip-utils';
+import { ExtraTimeSeriesEventData, RequiredTimeSeriesEventData } from '../timeSeries/TimeSeries';
+import './curation-history-tab.scss';
+
+export interface ICurationHistoryTabProps extends StoreProps {
+  historyData: HistoryList;
+}
+
+type HistoryTabData = {
+  record: HistoryRecord;
+  timeStamp: string;
+  admin: string;
+  location: string;
+  objectField?: string;
+};
+
+const CurationHistoryTab = observer(
+  ({ historyData, usersData, addUsersListener, historyTabStore, drugList, addDrugListListener }: ICurationHistoryTabProps) => {
+    const firebaseUsersPath = getFirebasePath('USERS');
+
+    const parsedHistoryData: HistoryTabData[] = useMemo(() => {
+      if (!historyData) {
+        return;
+      }
+
+      const data: HistoryTabData[] = [];
+      for (const history of Object.values(historyData)) {
+        try {
+          for (const record of history.records) {
+            if (typeof record.new === 'object') {
+              if (record.new['description']) {
+                data.push({
+                  record,
+                  timeStamp: history.timeStamp,
+                  admin: history.admin,
+                  location: record.location,
+                  objectField: 'description',
+                });
+              }
+              if (record.new['oncogenic']) {
+                data.push({
+                  record,
+                  timeStamp: history.timeStamp,
+                  admin: history.admin,
+                  location: record.location,
+                  objectField: 'oncogenic',
+                });
+              }
+              if (record.new['effect']) {
+                data.push({
+                  record,
+                  timeStamp: history.timeStamp,
+                  admin: history.admin,
+                  location: record.location,
+                  objectField: 'effect',
+                });
+              }
+            } else {
+              data.push({
+                record,
+                timeStamp: history.timeStamp,
+                admin: history.admin,
+                location: record.location,
+              });
+            }
+          }
+        } catch {
+          continue;
+        }
+      }
+
+      data.sort((data1, data2) => new Date(data2.timeStamp).getTime() - new Date(data1.timeStamp).getTime());
+      return data;
+    }, [historyData]);
+
+    const authorDropdownOptions = useMemo(() => {
+      const options = [];
+      if (usersData) {
+        for (const [email, data] of Object.entries(usersData)) {
+          options.push({ value: email, label: data.name || `[${email}]` });
+        }
+      }
+      return options;
+    }, [usersData]);
+
+    useEffect(() => {
+      const cleanupCallbacks = [];
+      cleanupCallbacks.push(addUsersListener(firebaseUsersPath));
+      cleanupCallbacks.push(addDrugListListener());
```

Comment:
Why do we need these two?

Reply:
The Users listener is to populate the author dropdown from the firebase users. The drugList is necessary to display the location correctly. Since the location comes with a list of drug uuids, I need the drug list to retrieve the drug name (this is all inside `formatLocation` and uses the `getTxName`) helper function